### PR TITLE
Italian Vat id and Tax id support

### PR DESCRIPTION
**Pull request by brainrepo**
Italian TaxId (Codice fiscale) https://it.wikipedia.org/wiki/Codice_fiscale
Italian VatId (Partita iva) https://it.wikipedia.org/wiki/Partita_iva

### CODE REVIEW

File: test/Faker/Provider/it_IT/CompanyTest.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    {
+        $faker = new Generator();
+        $faker->addProvider(new Company($faker));
+        $this->faker = $faker;
+    }
+
+    public function testIfTaxIdCanReturnData()
+    {
+        $vatId = $this->faker->vatId();
+        $this->assertRegExp('/^IT{12}[0-9]$/', $vatId);
```
Comment (EmanueleMinotto) [inline test/Faker/Provider/it_IT/CompanyTest.php]: I don't know exactly how the `regexify` method works, but this doesn't match an italian VAT :\

Comment (brainrepo) [inline test/Faker/Provider/it_IT/CompanyTest.php]: Ops it is a mistake i fix it now!
Tnks

**Comment by brainrepo** [comment]
I fix the mistake

File: src/Faker/Provider/it_IT/Company.php [hunk 2]
```diff
@@ -61,4 +61,14 @@ public function bs()
...
     }
+
+    /**
+     * Italian VAT number (Partita iva)
+     * @link https://it.wikipedia.org/wiki/Partita_IVA
+     * @return string
+     */
+    public static function vatId()
+    {
+        return static::regexify('/^IT[0-9]{12}$/');
```
Comment (EmanueleMinotto) [inline src/Faker/Provider/it_IT/Company.php]: Quoting from https://en.wikipedia.org/wiki/VAT_identification_number :
> 11 digits (the first 7 digits is a progressive number, the following 3 means the province of residence, the last digit is a check number)

File: src/Faker/Provider/it_IT/Company.php [hunk 3]
```diff
@@ -61,4 +61,14 @@ public function bs()
...
     }
+
+    /**
+     * Italian VAT number (Partita iva)
+     * @link https://it.wikipedia.org/wiki/Partita_IVA
+     * @return string
+     */
+    public static function vatId()
+    {
+        return static::regexify('/^IT[0-9]{11}$/');
```
Comment (fzaninotto) [inline src/Faker/Provider/it_IT/Company.php]: `regexify` is very slow, use `numerify` instead.

File: src/Faker/Provider/it_IT/Person.php [hunk 4]
```diff
@@ -85,4 +85,14 @@ public static function suffix()
...
     }
+
+    /**
+     * TaxCode (CodiceFiscale)
+     * @link https://it.wikipedia.org/wiki/Codice_fiscale
+     * @return string
+     */
+    public static function taxId()
+    {
+        return static::regexify('/^[A-Z]{6}[0-9]{2}[A-Z][0-9]{2}[A-Z][0-9]{3}[A-Z]$/');
```
Comment (fzaninotto) [inline src/Faker/Provider/it_IT/Person.php]: same remark as above

**Comment by fzaninotto** [comment]
You should add a mention of the new providers in the Readme, under the "Locales" section.

**Comment by brainrepo** [comment]
Travis fail in HHVM build

```
There was 1 failure:
1) Faker\Test\Provider\InternetTest::testIpv4NotLocalNetwork
Failed asserting that '1.156.0.249' does not match PCRE pattern "/\A1\./".
/home/travis/build/fzaninotto/Faker/test/Faker/Provider/InternetTest.php:113
```

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**